README.md

```diff
@@ -100,6 +100,38 @@ lock(resource: 'some_resource', skipIfLocked: true) {
 }
 ```
 
+#### Finding Examples
+
+*List all resources acquired by the current build*
+
+```groovy
+lock(label:'printer') {
+  echo findLocks().name
+}
+```
+
+*List all resources (locked or not) currently defined*
```

Comment:
Thinking of the repetitive "locked or not" comment, maybe filters for lock states (not-locked - `build:'none'` perhaps?, reserved, stolen, ephemeral/persistent...) could be useful.

Reply:
My initial concern was to go in a feature creep that would add complexity without fulfilling actual use cases. Do we want to search in notes? what about `reservedTimestamp` - and do we want to allow before / after searches, etc. I don't mind adding a few, but I think the missing ones will be discovered and added through pull requests explaining which use case the new find option is fulfilling. 

For `reservedBy`, I think it opens an interesting case where users can reserve resources and then execute jobs that work on their reserved resources, but apart from unreserving them (ex. with the update step), but it would be difficult to allow a step lock all the resources reserved by a user unless we allow locking resources reserved by a user _when the job is being triggerred by that user_ for example.

For stolen, ephemeral, these are easy to add, but I have the same concern about mapping these to actual use cases.

What I am hoping is that the combo of findLock/updateLock and the advanced lock filters will resolved most of the us cases by using labeling. If you really need to work on your reserved locks, `findLocks` on all locks, look for the ones you reserved, update the labels (ex. `reserved_by_<username>`) and then you can lock these specific locks by label.

I do have a much narrower view on how this plugin might be used by the rest of the community, so I'll rely on your opinion on which fields are worth adding to the findLock step